File: FrontEnd/ogani/src/app/_class/order-detail.ts
export class OrderDetail {
    name !: string;
    price !: number;
    quantity !: number;
    productId?: number; // ID sản phẩm để track tồn kho

    // SubTotal là giá trị được tính toán, không cần gửi đến backend
    get subTotal(): number {
        return this.price * this.quantity;
    }

    constructor(name: string = '', price: number = 0, quantity: number = 0, productId?: number) {
        this.name = name;
        this.price = price;
        this.quantity = quantity;
        this.productId = productId;
    }
}

File: BackEnd/ogani/src/main/java/com/example/ogani/entity/OrderDetail.java
package com.example.ogani.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonBackReference;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "order_details")
public class OrderDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;

    private Long price;
    private Integer quantity;

    private Long subTotal;

    // ID sản phẩm để truy vết và quản lý tồn kho
    @jakarta.persistence.Column(name = "product_id")
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "order_id", nullable = false)
    @JsonBackReference
    private Order order;
}
